tests/python/gaia-ui-tests/gaiatest/apps/settings/regions/call_settings.py

```diff
@@ -0,0 +1,211 @@
+
+    @property
+    def vm_screen_element(self):
+        return self.marionette.find_element(*self._voicemail_page_locator)
+
+    @property
+    def fd_screen_element(self):
+        return self.marionette.find_element(*self._fd_page_locator)
+
+    @property
+    def fd_simpin_screen_element(self):
+        return self.marionette.find_element(*self._sim_pin_page_locator)
+
+    @property
+    def auth_screen_element(self):
+        return self.marionette.find_element(*self._auth_number_page_locator)
+
+    @property
+    def cf_screen_element(self):
+        return self.marionette.find_element(*self._cf_page_locator)
+
+    @property
+    def cf_always_screen_element(self):
+        return self.marionette.find_element(*self._cf_always_page_locator)
+
+    @property
+    def cf_busy_screen_element(self):
+        return self.marionette.find_element(*self._cf_busy_page_locator)
+
+    @property
+    def cf_unans_screen_element(self):
+        return self.marionette.find_element(*self._cf_unanswered_locator)
+
+    @property
+    def cf_unreach_screen_element(self):
+        return self.marionette.find_element(*self._cf_unreachable_locator)
+
+    @property
+    def cb_screen_element(self):
+        return self.marionette.find_element(*self._cb_page_locator)
+
+    @property
+    def cb_passcode_screen_element(self):
+        return self.marionette.find_element(*self._passcode_page_locator)
+
+    def tap_sim_1(self):
+        element = self.marionette.find_element(*self._sim_1_locator)
+        Wait(self.marionette).until(expected.element_displayed(element)and expected.element_enabled(element))
+        element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._cs_page_locator))
+
+    def tap_voicemail(self):
+        element = self.marionette.find_element(*self._voicemail_locator)
+        Wait(self.marionette).until(expected.element_displayed(element)and expected.element_enabled(element))
+        element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._voicemail_number_locator))
+
+    def tap_caller_id_selection(self):
+        menu_item = self.marionette.find_element(*self._caller_id_menu_item_locator)
+        Wait(self.marionette).until(lambda m: not menu_item.get_attribute('aria-disabled'))
+
+        element = self.marionette.find_element(*self._caller_id_selector_locator)
+        Wait(self.marionette).until(expected.element_displayed(element)and expected.element_enabled(element))
+        element.tap()
+        self.marionette.switch_to_frame()
+        Wait(self.marionette).until(expected.element_displayed(*self._caller_id_confirm_button_locator))
+
+    def confirm_caller_id_selection(self):
+        element = self.marionette.find_element(*self._caller_id_confirm_button_locator)
+        Wait(self.marionette).until(expected.element_displayed(element)and expected.element_enabled(element))
+        element.tap()
+        self.apps.switch_to_displayed_app()
```

Comment:
Perhaps it's going wrong here and you need to add a wait.until settings app is the displayed_app?